src/Common/Versioning/Conventions/ActionMethodResolver.cs

```diff
@@ -0,0 +1,64 @@
+﻿#if WEBAPI
+namespace Microsoft.Web.Http.Versioning.Conventions
+#else
+namespace Microsoft.AspNetCore.Mvc.Versioning.Conventions
+#endif
+{
+    using System;
+    using System.Diagnostics.Contracts;
+    using System.Linq;
+    using System.Reflection;
+#if WEBAPI
+    using System.Web.Http;
+#endif
+
+    static class ActionMethodResolver
+    {
+        internal static MethodInfo Resolve( Type controllerType, string methodName, Type[] argumentTypes )
+        {
+            Contract.Requires( controllerType != null );
+            Contract.Requires( !string.IsNullOrEmpty( methodName ) );
+
+            var methods = controllerType.GetRuntimeMethods().Where( m => m.Name == methodName && IsAction( m ) ).ToArray();
+
+            switch ( methods.Length )
```

Comment:
Reasonable.

Where applicable, I try apply my _would be_ performance skills, however small they may be. A `switch` state will result in jump statements that are more efficent that using `if` statement which are a branch. A `switch`, `break`, `continue`, and a few other keywords are really still `goto` statements under the hood. If I didn't have to use `methods` multiple times, I would not have converted it to an array. Since I did, I'm able to optimize the use of the `Length` (e.g. count) property; otherwise, I almost certainly would have used an `if` here.

In terms of cases, you're right - there is a `default`. I've gotten the habit of avoiding nested blocks; especially if they're more than 1-2 lines. It's purely stylistic and I _feel_ it's easier to read. I do the same thing with `if` statements. If I can omit the `else` branch or otherwise short-circuit to prevent a large intended block, I tend to prefer that.